rskj-core/src/main/java/co/rsk/peg/federation/PendingFederation.java

```diff
@@ -170,4 +172,52 @@ public int hashCode() {
         // well-defined hashCode()
         return Objects.hash(getBtcPublicKeys());
     }
+
+    /**
+     * A pending federation is serialized as the
+     * public keys conforming it.
+     * A list of btc public keys is serialized as
+     * [pubkey1, pubkey2, ..., pubkeyn], sorted
+     * using the lexicographical order of the public keys
+     * (see BtcECKey.PUBKEY_COMPARATOR).
+     * This is a legacy format for blocks before the Wasabi
+     * network upgrade.
+     */
+    public byte[] serializeOnlyBtcKeys() {
+        return serializeBtcPublicKeys(this.getBtcPublicKeys());
+    }
+    private static byte[] serializeBtcPublicKeys(List<BtcECKey> keys) {
+        List<byte[]> encodedKeys = keys.stream()
+            .sorted(BtcECKey.PUBKEY_COMPARATOR)
+            .map(key -> RLP.encodeElement(key.getPubKey()))
+            .collect(Collectors.toList());
+        return RLP.encodeList(encodedKeys.toArray(new byte[0][]));
+    }
+
+    /**
+     * A pending federation is serialized as the
+     * list of its sorted members serialized.
+     * A FederationMember is serialized as a list in the following order:
+     * - BTC public key
+     * - RSK public key
+     * - MST public key
+     * All keys are stored in their COMPRESSED versions.
+     */
+    public byte[] serialize() {
+        List<byte[]> encodedMembers = this.getMembers().stream()
+            .sorted(FederationMember.BTC_RSK_MST_PUBKEYS_COMPARATOR)
+            .map(PendingFederation::serializeMember)
+            .collect(Collectors.toList());
+        return RLP.encodeList(encodedMembers.toArray(new byte[0][]));
+    }
+
+    private static byte[] serializeMember(FederationMember federationMember) {
```

Comment:
Maybe this could be part of FederationMember class? So each class knows how to serialize itself

Reply:
good idea!